### DISCUSSION TITLE
Any tutorials on contributing to elementary native apps OS?

### DISCUSSION THREAD

**Discussion by Borwe** [elementary/triage#113] [Q&A]
I am asking this, because there is no clear way of how one can contribute, to say wingpanel locally, and see results, expecially if the build is targeting Ubuntu 20.04 LTS like eOS 6, and all I got is Elementary Os 5.1.7 hera, expecially if the code involves UI stuff.

**Comment by ryanprior** [comment]
Hey @Borwe I wrote a tutorial that covers some aspects: https://www.ryanprior.com/posts/changing-power-settings-in-elementary-os/

Lots more things that aren't covered there are covered in the OS development contributing guide: https://elementary.io/docs/code/os-dev#elementary-sdk

**Reply by Borwe** [reply]
> Hey @Borwe I wrote a tutorial that covers some aspects: https://www.ryanprior.com/posts/changing-power-settings-in-elementary-os/
> 
> Lots more things that aren't covered there are covered in the OS development contributing guide: https://elementary.io/docs/code/os-dev#elementary-sdk

This is cool info, but was wondering as to how I can update the app itself while also having a similar instance running locally? EG: instead of using terminal to add the turn off screen after 1 minute option, but to actually put it in the switchboard-plug-power as an option on UI ... as in what's the standard rule when it comes to such? I believe one would be building, running and testing multiple times when contributing to eOS, and with the boost to 20.04 base, was curious how people are handling this when contributing to eOS 6 on a 18.04 base?

**Reply by hanaral** [reply]
You can't have two instances of the app running locally, use a VM or multiboot with the daily. I'd recommend running locally, but on a spare drive you have.

> EG: instead of using terminal to add the turn off screen after 1 minute option, but to actually put it in the switchboard-plug-power as an option on UI ... as in what's the standard rule when it comes to such?

Are you asking about design guidelines development or actual elementary OS system component development?
If it's the latter, you use `git` to 'fork' the repo to your device and then make the changes from Code or whatever editor you use. Once you're done you can open a 'pull request' to merge your changes (I think git can open the pull request in yourbrowser if you make it from command line)
You should be able to get this info from the contributing website

**Comment by librafrog** [comment]
Maybe some kind of _sandbox_ in which users of the OS can offer mostly theme, colors aka design change to different built-in apps 🔢
